### ISSUE TITLE
[docdb] CHECK failure in yb::tablet::MvccManager::DoGetSafeTime in perf itest

### ISSUE THREAD

**Issue by mbautin** [yugabyte/yugabyte-db#176]
Jira Link: [DB-2083](https://yugabyte.atlassian.net/browse/DB-2083)
Happened on perf itest two times in the same run.

https://gist.githubusercontent.com/mbautin/1295f2937ad2f4670e93a81dfb98d447/raw
https://gist.githubusercontent.com/mbautin/92a8952b0d6bb39a6a41eb683c4bec4f/raw

```
F20180308 19:47:21 ../../../../../src/yb/tablet/mvcc.cc:224] Check failed: result >= has_lease ? max_safe_time_returned_with_lease_ : max_safe_time_returned_without_lease_ ({ physical: 1520538441080801 } vs. { physical: 1520538441080801 logical: 1215 }) T 126c036862ae4a1a9a45a7cc3e2bd1c3 has_lease=1, ht_lease={ physical: 1520538442921977 }, max_ht_lease_seen_={ physical: 1520538442921977 }, last_replicated={ physical: 1520538428430751 logical: 8 }, clock_->Now()={ physical: 1520538441081271 }
```

and then

```
F20180308 20:04:08 ../../../../../src/yb/tablet/mvcc.cc:224] Check failed: result >= has_lease ? max_safe_time_returned_with_lease_ : max_safe_time_returned_without_lease_ ({ physical: 1520539448153492 } vs. { physical: 1520539448153492 logical: 939 }) T 0794ef38cd8344b79f478bd09a66eb95 has_lease=1, ht_lease={ physical: 1520539450011114 }, max_ht_lease_seen_={ physical: 1520539450011114 }, last_replicated={ physical: 1520539374036435 logical: 170 }, clock_->Now()={ physical: 1520539448153623 }
```

**Comment by amitanandaiyer** [comment]
may be related to #2571

**Comment by kmuthukk** [comment]
@spolitov :

Is this issue addressed by #2571 fixes (two of them which went in)?

https://github.com/yugabyte/yugabyte-db/commit/401e821a52840892b77fb1c7dd771454ff060e7a

https://github.com/yugabyte/yugabyte-db/commit/c053ba3ba7aca533bf7537795e570436683cc462
